### ISSUE TITLE
utlfetch fails with ssl.SSLError: [SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:877)

### ISSUE THREAD

**Issue by dmarjanovic** [ifduyue/urlfetch#48]
Any idea how to fix/workaround/diagnose/debug this? Is there anything specific about `urlfetch` when it comes to HTTPS+proxy that would cause issue to appear here but not in other python libs? How is possible that protocol is unknown? It looks like it has something to do with proxy & SSL and the way how `urlfetch` is handling it but can't get any more details about error from underlaying SSL even when using debugger. The proxy is properly configured. HTTP works always fine. *HTTPS fails when proxy is used* but works just fine for all internal URLs that are not proxied. Please note `wget` and `curl` do work just fine as well as `urllib` and `urllib2`.
```
$ python
Python 3.6.8 (default, Apr  2 2020, 13:34:55) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-39)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>
>>> import urlfetch
>>> urlfetch.get('https://dl.google.com/')
Traceback (most recent call last):
  File "/home/mmmm/Code/google/android-emulator-container-scripts/venv/lib64/python3.6/site-packages/urlfetch-1.2.1-py3.6.egg/urlfetch.py", line 755, in request
    conn.request(method, request_url, data, reqheaders)
  File "/usr/lib64/python3.6/http/client.py", line 1254, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib64/python3.6/http/client.py", line 1300, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/lib64/python3.6/http/client.py", line 1249, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib64/python3.6/http/client.py", line 1036, in _send_output
    self.send(msg)
  File "/usr/lib64/python3.6/http/client.py", line 974, in send
    self.connect()
  File "/usr/lib64/python3.6/http/client.py", line 1415, in connect
    server_hostname=server_hostname)
  File "/usr/lib64/python3.6/ssl.py", line 365, in wrap_socket
    _context=self, _session=session)
  File "/usr/lib64/python3.6/ssl.py", line 773, in __init__
    self.do_handshake()
  File "/usr/lib64/python3.6/ssl.py", line 1033, in do_handshake
    self._sslobj.do_handshake()
  File "/usr/lib64/python3.6/ssl.py", line 645, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:877)
```

Couple of example of other python libs working just fine:
```
>>> import urllib
>>> f = urllib.FancyURLopener()
>>> f.retrieve("https://dl.google.com", "test.html")
```
or:
```
>>> import urllib2
>>> response = urllib2.urlopen('https://dl.google.com/')
>>> html = response.read()
```

This is discovered while using `emu-docker` script from Google team and also reported https://github.com/google/android-emulator-container-scripts/issues/159 but as mentioned there seems to be an issue with `urlfetch` itself.

Thank you.

/cc @ifduyue

**Comment by ifduyue** [comment]
@dmarjanovic did you mean http_proxy and https_proxy? What's the value?

**Comment by ifduyue** [comment]
@dmarjanovic Please check if commit 19050d0 fixed this issue: 

```
# under google/android-emulator-container-script directory
$ source ./configure.sh
$ pip install -U  git+https://github.com/ifduyue/urlfetch.git@v1.x.x
$ emu-docker list
```

**Comment by dmarjanovic** [comment]
@ifduyue yes, the fix worked. Do you know when can the fix be included in new release? Thank you very much. 👍

**Comment by ifduyue** [comment]
@dmarjanovic bugfix version 1.2.2 uploaded to pypi https://pypi.org/project/urlfetch/1.2.2/

**Comment by dmarjanovic** [comment]
@ifduyue fix verified in 1.2.2! Thank you very much. 👍
